### PR TITLE
Core/Movement: Fix fleeing speed to 66% run speed

### LINKED ISSUE #21899: Creature flee speed when "run-away" triggers is too fast!
**Description: Creature flee speed when "run-away" triggers is too fast!**

When fighting mobs that run away when under a certain health level, more often than not they run away at 100%+ full speed. Sometimes they "hobble"/walk away as intended, or do the running animation but look snared.

**Current behaviour: Runs away at 100% speed**

Mob runs away too quickly to rally nearby enemies for support.

**Expected behaviour:**

Mob speed should be closer to that of walking than full on sprinting AFAIK. This does not feel blizzlike.

**Steps to reproduce the problem:**

1. Blackrock Sentry - 4065 (all blackrock), Redridge Mystic - 430 (all redridge gnolls).
2. Fight mob until it triggers creature flee at X hp.
3. Be annoyed that it's already out of your range and running away full speed pulling more mobs that do the same which exponentially grows as more start fleeing full speed in all directions. Die. Repeat.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. 88130d86b0ab 2018-03-29 20:18:52 +0200 (3.3.5 branch) (Win64, R
elease, Static)
Using SSL version: OpenSSL 1.1.0h  27 Mar 2018 (library: OpenSSL 1.1.0g  2 Nov 2
017)
Using Boost version: 1.65.1
Using MySQL version: 5.6.39
Using CMake version: 3.11.0
Compiled on: Windows 10.0.16299
Automatic database updates are disabled for all databases!
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in data/maps. Total size: 252191207 bytes
vmaps directory located in data/vmaps. Total size: 588247501 bytes
mmaps directory located in data/mmaps. Total size: 2152621724 bytes
Using enUS DBC Locale as default. All available DBC locales: enUS
Using World DB: TDB 335.64

**Operating system: Windows 10**

*** Tried googling this issue and searching the issues on here, found nothing that seemed related. ***

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Fix fleeing speed to 66% run speed as custom spline speed

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #21899 , replaces #23990

**Tests performed:** (Does it build, tested in-game, etc.)
I don't remember, someone will have to test it

**Known issues and TODO list:** (add/remove lines as needed)

- None

### CODE REVIEW

File: src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp [hunk 1]
```diff
@@ -176,6 +176,7 @@ void FleeingMovementGenerator<T>::SetTargetLocation(T* owner)
     Movement::MoveSplineInit init(owner);
     init.MovebyPath(_path->GetPath());
     init.SetWalk(false);
+    init.SetVelocity(owner->GetSpeed(MOVE_RUN) * 0.66f);
```
Comment (Shauren) [inline src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: This movement generator is also used for regular fears (including targeting players), don't set it unconditionally

Comment (ryancheung) [reply src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: I have to mention that some fear spells have aura effect that reduces speed, Intimidating Shout for instance. We may not need to change the flee speed manually.

Comment (jackpoz) [reply src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: Maybe check unit_fleeing or fear aura count ?

Comment (ryancheung) [reply src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: Maybe add a fear flag in `Unit::SetFeared` ?

Comment (Shauren) [reply src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: How about adding another arg to `MotionMaster::MoveFleeing`? `bool reducedSpeed`, set to true if inside `Unit::SetFeared` there are no fear auras present (its already checked there for other reasons) and in SAI flee action

Comment (ryancheung) [reply src/server/game/Movement/MovementGenerators/FleeingMovementGenerator.cpp]: I just did that https://github.com/TrinityTBC/core/commit/00df0e39a7ff400463b7ec96716c4009ca03a1c7, lol

**Comment by jackpoz** [comment]
PR updated with latest changes requested (thanks @ryancheung for the reference commit, especially the the template specialization changes)

**Comment by Killyana** [comment]
I checked many videos from classic wow, and made many tests in retail:
- Once the npc flee or call for help, its speed will be reduced until it dies or reset (evade)
- If the npc call for help once it reach the position it will move randomly (5yards) before making other creatures aggro.

**Comment by jackpoz** [comment]
@Killyana that seems to be unrelated to the speed itself, would it be ok to merge this PR first and deal with whatever leftover issue another time ?

**Comment by Killyana** [comment]
For The 2 first issue yes, even restoring the speed on evade looks easy to do.
But the combat behavior when the pnc is fleeing is caused by this PR.

**Comment by jackpoz** [comment]
as more and more changes/issue keep piling up and this was supposed to fix only the Seeking Help case, I might merge the single commit https://github.com/TrinityCore/TrinityCore/pull/23998/commits/b931fc9dca13475805d347220045617a3f31992f and leave everything else the way it is so someone else can try fix that

**Comment by jackpoz** [comment]
The issue that https://github.com/TrinityCore/TrinityCore/pull/23990 was trying to fix has been implemented in https://github.com/TrinityCore/TrinityCore/commit/862d47394b8c879f42902f03ecd112b65ff631c2 . If anyone is interested, feel free to fix the remaining fleeing speed issues

**Pull request closed without merging**